website/docs/d/search_deployment.html.markdown

```diff
@@ -1,37 +1,81 @@
 
-## Example Usage
 
+## Example Usages
 ```terraform
-data "mongodbatlas_search_deployment" "test" {
-    project_id = "<PROJECT_ID>"
-    cluster_name = "<CLUSTER_NAME>"
+resource "mongodbatlas_project" "example" {
+  name   = "project-name"
+  org_id = var.org_id
+}
+
+resource "mongodbatlas_advanced_cluster" "example" {
+  project_id   = mongodbatlas_project.example.id
+  name         = "ClusterExample"
+  cluster_type = "REPLICASET"
+
+  replication_specs {
+    region_configs {
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+      provider_name = "AWS"
+      priority      = 7
+      region_name   = "US_EAST_1"
+    }
+  }
+}
+
+resource "mongodbatlas_search_deployment" "example" {
+  project_id   = mongodbatlas_project.example.id
+  cluster_name = mongodbatlas_advanced_cluster.example.name
+  specs = [
+    {
+      instance_size = "S20_HIGHCPU_NVME"
+      node_count    = 2
+    }
+  ]
+}
+
+data "mongodbatlas_search_deployment" "example" {
+  project_id   = mongodbatlas_search_deployment.example.project_id
+  cluster_name = mongodbatlas_search_deployment.example.cluster_name
+}
+
+output "mongodbatlas_search_deployment_id" {
+  value = data.mongodbatlas_search_deployment.example.id
 }
 ```
 
-## Argument Reference
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `cluster_name` (String) Label that identifies the cluster to return the search nodes for.
+- `project_id` (String) Unique 24-hexadecimal digit string that identifies your project.
 
-* `project_id` - (Required) The unique identifier for the [project](https://docs.atlas.mongodb.com/organizations-projects/#std-label-projects) that contains the specified cluster.
-* `cluster_name` - (Required) The name of the cluster containing a search node deployment.
+### Read-Only
 
-## Attributes Reference
+- `id` (String) Unique 24-hexadecimal digit string that identifies the search deployment.
+- `specs` (Attributes List) List of settings that configure the search nodes for your cluster. (see [below for nested schema](#nestedatt--specs))
+- `state_name` (String) Human-readable label that indicates the current operating condition of this search deployment.
 
-* `specs` - List of settings that configure the search nodes for your cluster. See [specs](#specs).
-* `state_name` - Human-readable label that indicates the current operating condition of this search node deployment.
+<a id="nestedatt--specs"></a>
+### Nested Schema for `specs`
 
-### Specs
-* `instance_size` - (Required) Hardware specification for the search node instance sizes. The [MongoDB Atlas API](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Atlas-Search/operation/createAtlasSearchDeployment) describes the valid values. More details can also be found in the [Search Node Documentation](https://www.mongodb.com/docs/atlas/cluster-config/multi-cloud-distribution/#search-tier).
```

Comment:
I would preserve this more detailed description of the instance size attribute as it also references valid values, we can adjust the schema description fields and also include a [description override config](https://developer.hashicorp.com/terraform/plugin/code-generation/openapi-generator#overriding-attribute-descriptions) in the generator_config.yml file.

Reply:
I was planning to modify this in the open api spec but this may be better. I will also update our doc to make sure this is documented